server/src/v2/modules/db.js

```diff
@@ -9,36 +9,85 @@
 
 import Database from 'better-sqlite3';
 
-const DROP_SECRET_TABLE_QUERY = `
-DROP TABLE IF EXISTS
-secret;
-`;
-
 /*
-  Create Table Query
-  Table name: secret
-    id: The primary key id for each secret
-    secret_text: The string of the secret
-    passphrase: The string of the hashed password
-    expiryDate: The date and time of when the secret should expire
-    method: A code to indicate which method was used for encryption
-    access_failed_attempts: The current number of failed attempts accesssing the secret
+Global variable for the database file
 */
-const CREATE_TABLE_QUERY = `
+
+var TABLE_NAME = "Secret";
+var databasePath = "./secrets.db"
+var databaseFile;
+
+function initialise() {
+  //databaseFile = new Database("secrets.db");
+  databaseFile = new Database(databasePath, { verbose: console.log });
+  const initialisationFail = TRUE;
+  databaseFile.transaction.exclusive(() => { //Lock out all database functions when initialising, and put into transaction
+    const dropStatement = databaseFile.prepare('DROP TABLE IF EXISTS ?');
+    dropStatement.run(TABLE_NAME);
+    console.log("Cleared Table ${TABLE_NAME} (if exists).");
+    /*
+      Create Table Query
+      Table name: secret
+        id: The primary key id for each secret
+        secret_text: The string of the secret
+        passphrase: The string of the hashed password
+        expiryDate: The date and time of when the secret should expire
+        method: A code to indicate which method was used for encryption
+        access_failed_attempts: The current number of failed attempts accesssing the secret
+    */
+    const createStatement = databaseFile.prepare(`
 CREATE TABLE
-'secret'(
+'?'(
     id TEXT PRIMARY KEY NOT NULL,
     secret_text TEXT NOT NULL,
     passphrase TEXT NOT NULL,
     expiry_date DATE NOT NULL,
     method TEXT NOT NULL,
     access_failed_attempts INT NOT NULL DEFAULT 0
     )
-;
-`;
-const INSERT_SECRET_QUERY = `
+`);
+    createStatement.run(TABLE_NAME);
+    console.log("Created New Table ${TABLE_NAME}.")
+    initialisationFail = FALSE;
+  });
+  if (initialisationFail){
+    console.log("Database initialised successfully.");
+  }
+  else {
+    console.error("Could not initialise database! If you do not know why this error occurred, try deleting the database file, located at '${databasePath}' and trying again later.");
+  }
+}
+
+initialise();
+
+function getStatement(preparedStatement, statementParams) {
+  //Returns data
+  let row;
+  try {
+    row = preparedStatement.get(statementParams);
+  }
+  catch (err) {
+    return { data: null, code: 500, human_code: `failure, ${err}` };
+  }
+  return { data: row, code: 200, human_readable_code: "Success" };
```

Comment:
Please change the `code` attribute to `success` where it returns either `true` or `false`. Using HTTP status codes for a binary "yes this succeeded", or "no this failed" isn't really appropriate in this context.

https://github.com/Group4-3/ESSW/blob/main/server/src/v2/routes/secret/submit.js#L63 will need to be updated to `if (transaction.status === true) { ... }`

`human_code` and `human_readable_code` should be removed. Error returns should inlcude an `error` attribute instead which contains the SQLite error message. This isn't actually used anywhere at the moment so this PR should not be concerned with changing that.

Reply:
Addressed in [f471432](https://github.com/Group4-3/ESSW/pull/18/commits/f4714328a204743ca88a94eb145439cf6fffec8e).